ports/rp2/mphalport.c

```diff
@@ -266,15 +266,13 @@ void soft_timer_init(void) {
 }
 
 void mp_wfe_or_timeout(uint32_t timeout_ms) {
-    soft_timer_entry_t timer;
-
-    // Note the timer doesn't have an associated callback, it just exists to create a
-    // hardware interrupt to wake the CPU
-    soft_timer_static_init(&timer, SOFT_TIMER_MODE_ONE_SHOT, 0, NULL);
-    soft_timer_insert(&timer, timeout_ms);
-
-    __wfe();
+    best_effort_wfe_or_timeout(delayed_by_ms(get_absolute_time(), timeout_ms));
```

Comment:
It might be best to stick with our custom implementation here, if that's possible.  I'm still a bit concerned about issues like https://github.com/raspberrypi/pico-sdk/issues/2118

Reply:
I don't think your custom implementation will work and allow the device to go to sleep on rp2350 for reasons we already discussed? https://github.com/raspberrypi/pico-sdk/issues/1812

Reply:
@dpgeorge I think that issue can impact machine.Timer if you run the device at a clock speed above 160MHz via machine.freq and the alarm or repeated timer callback takes <1us. I will leave the commit in place for the moment. If you want me to put back the soft timer, I'll remove the changes I've made in this area. (Including the "fixes" for light sleep?). 

One possible workaround (while waiting for a fix to the sdk) would be to add a 1us busy wait to the alarm_callback function. Can you let me know what you prefer me to do?

Reply:
I'd rather be conservative here and not change anything around the alarm pool until that pico-sdk issue is fixed (or we can be confident the issue does not affect MicroPython).  Can we please take that change out of this PR and just make this PR about adding Pico 2 W support?  Then we can have a separate PR with the lightsleep fix (you can open that now to not lose track of it) and test and apply that separately.

Reply:
I've dropped the timer changes.